codegenerator/cli/templates/static/codegen/src/EventModLookup.res

```diff
@@ -0,0 +1,109 @@
+open Belt
+type eventMod = module(Types.InternalEvent)
+
+type errorKind = WildcardSighashCollision | Duplicate
+type eventError = {eventMod: eventMod, errorKind: errorKind}
+module ContractEventMods = {
+  type t = {
+    all: array<eventMod>,
+    byContractName: dict<eventMod>,
+  }
+
+  let empty = () => {
+    all: [],
+    byContractName: Js.Dict.empty(),
+  }
+
+  let isWildcard = (eventMod: eventMod) => {
+    let module(Event) = eventMod
+    Event.handlerRegister->Types.Handlers.Register.getEventOptions->(v => v.Types.Handlers.wildcard)
+  }
+
+  let hasWildcardCollision = (eventModA, eventModB) => {
+    eventModA->isWildcard || eventModB->isWildcard
+  }
+
+  let set = ({all, byContractName}: t, eventMod: eventMod) => {
+    let module(Event) = eventMod
+    switch byContractName->Js.Dict.get(Event.contractName) {
+    | Some(_) => Error({eventMod, errorKind: Duplicate})
+    | None =>
+      if all->Array.some(hasWildcardCollision(_, eventMod)) {
+        Error({eventMod, errorKind: WildcardSighashCollision})
+      } else {
+        all->Js.Array2.push(eventMod)->ignore
+        byContractName->Js.Dict.set(Event.contractName, eventMod)
+        Ok()
+      }
+    }
+  }
+
+  let getByContractName = ({byContractName}, ~contractName) => {
+    byContractName->Js.Dict.get(contractName)
+  }
+
+  let findWildcard = ({all}) => {
+    all->Js.Array2.find(event => event->isWildcard)
+  }
```

Comment:
With the new design you can add `wildcard: option<eventMod>` field to the `ContractEventMods.t` type, which will simplify the `hasWildcardCollision` and allow getting rid of the `findWildcard` function